functions/New-DbaAgentJob.ps1

```diff
@@ -0,0 +1,476 @@
+		[object[]]$SqlServer,
+        [Parameter(Mandatory = $false)]
+        [System.Management.Automation.PSCredential]$SqlCredential,
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$JobName,
+        [Parameter(Mandatory = $false)]
+        [switch]$Disabled,
+        [Parameter(Mandatory = $false)]
+        [string]$Description,
+        [Parameter(Mandatory = $false)]
+        [int]$StartStepId,
+        [Parameter(Mandatory = $false)]
+        [string]$CategoryName,
+        [Parameter(Mandatory = $false)]
+        [int]$CategoryId,
+        [Parameter(Mandatory = $false)]
+        [string]$OwnerLoginName,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet(0, "Never", 1, "OnSuccess", 2, "OnFailure", 3, "Always")]
+        [object]$EventLogLevel = '',
+        [Parameter(Mandatory = $false)]
+        [ValidateSet(0, "Never", 1, "OnSuccess", 2, "OnFailure", 3, "Always")]
+        [object]$EmailLevel = '',
+        [Parameter(Mandatory = $false)]
+        [ValidateSet(0, "Never", 1, "OnSuccess", 2, "OnFailure", 3, "Always")]
+        [object]$NetsendLevel = '',
+        [Parameter(Mandatory = $false)]
+        [ValidateSet(0, "Never", 1, "OnSuccess", 2, "OnFailure", 3, "Always")]
+        [object]$PageLevel = '',
+        [Parameter(Mandatory = $false)]
+        [ValidateScript({
+            if(($EmailLevel -in 1, "OnSuccess", 2, "OnFailure", 3, "Always") -and ($_.Length -ge 1))
+            {
+                $true
+            }
+            else
+            {
+                Throw "Please set the e-mail operator when the e-mail level parameter is set."
+            }
+        })]
+        [string]$EmailOperatorName,
+        [Parameter(Mandatory = $false)]
+        [ValidateScript({
+            if(($NetsendLevel -in 1, "OnSuccess", 2, "OnFailure", 3, "Always") -and ($_.Length -ge 1))
+            {
+                $true
+            }
+            else
+            {
+                Throw "Please set the netsend operator when the netsend level parameter is set."
+            }
+        })]
+        [string]$NetsendOperatorName,
+        [Parameter(Mandatory = $false)]
+        [ValidateScript({
+            if(($PageLevel -in 1, "OnSuccess", 2, "OnFailure", 3, "Always") -and ($_.Length -ge 1))
+            {
+                $true
+            }
+            else
+            {
+                Throw "Please set the page operator when the page level parameter is set."
+            }
+        })]
+        [string]$PageOperatorName,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet(0, "Never", 1, "OnSuccess", 2, "OnFailure", 3, "Always")]
+        [object]$DeleteLevel = '',
+        [bool]$Force,
+        [switch]$Silent
+    )
+
+    BEGIN
+    {
+        # Check the input for the job id or job name
+        if($JobName.Length -lt 1)
+        {
+            Stop-Function -Message "Please enter a job name." -Silent $Silent -InnerErrorRecord $_ -Target $JobName
```

Comment:
General note on testing and the silent parameter:
I have found the inheritance issue not working perfectly _in tests_ when importing the .psm1 and the test function not being a part of the module (like pasted directly or imported separately. In this case the `$Silent` parameter may not be passed correctly. This can be worked around by either of these two methods:
 - Testing as part of module (Just paste new files in the correct folder, then rmo/ipmo)
 - Defining a `$Silent = $false` variable on the commandline